### DISCUSSION TITLE
Forward pipe operator

### DISCUSSION THREAD

**Discussion by lostmsu** [dotnet/csharplang#5315] [Language Ideas] [answered]
`|>` from F# might work with left-then-right argument computation order. I can't remember any scenarios where it would be a valid C# syntax. So it should be easy to parse that sequence as an individual token.

The use case is method chaining, that does not involve instance methods.

A simple use case is error checking:
```csharp
var validResult = GetPotentiallyInvalidValue()
                  |> EnsureIsValid();
```

Slighly more concrete use case is ML and TorchSharp in particular:
```csharp
var nn = torch.Input(10)
      |> torch.Linear(256)
      |> torch.Relu()
      |> torch.Linear(256)
      |> torch.Relu()
      |> torch.Linear(1);
```

Relevant [discussion of proposed JavaScript feature](https://news.ycombinator.com/item?id=28916775) with some convincing arguments for it.

**Comment by RikkiGibson** [comment, marked as answer]
#74
#96

**Comment by lostmsu** [comment]
Rant: It is slightly confusing, that searching issues does not bring up existing discussions. Not many are aware of discussions, and in my OSS project we basically decided to just use issue tags for this purpose for this exact reason.
